### PR TITLE
Fix some problems with debug mode and Intel compiler

### PR DESCRIPTION
**Pull request by ofuhrer**
Runs still don't work with Intel compiler in debug mode. These changes introduce the possibility to run multiple executables built on Piz Daint through the c12 and c48 checks (e.g. fv3_64bit.exe and fv3_64bit_debug.exe). Also, some fixes in the Fortran code which make the Intel compiler trip with debugging options on are done.

### CODE REVIEW

File: FV3/io/FV3GFS_io.F90 [hunk 1]
```diff
@@ -1593,7 +1593,7 @@ subroutine sfc_prop_override(Sfcprop, Grid, Atm_block, Model, fv_domain)
     type (domain2d),           intent(in)    :: fv_domain
     !--- local variables
     integer :: i, j, k, ix, lsoil, num, nb
-    integer :: isc, iec, jsc, jec, npz, nx, ny, ios
```
Comment (rheacangeo) [inline FV3/io/FV3GFS_io.F90:1596]: intel doesn't like declared but unused variables, I am guessing?

File: FV3/io/FV3GFS_io.F90 [hunk 2]
```diff
@@ -1637,6 +1637,9 @@ subroutine sfc_prop_override(Sfcprop, Grid, Atm_block, Model, fv_domain)
 
     call mpp_error(NOTE, "Calling sfc_prop_override")
 
+    isc = Atm_block%isc
```
Comment (rheacangeo) [inline FV3/io/FV3GFS_io.F90:1640]: wow this wasn't set before? how did ideal_sst work at all?

**Review by rheacangeo — COMMENTED** [review]
lgtm -- my main question is about the changes to the coarsening code, but if Spencer is ok with it than it's good to go. Most of my comments are just wondering if I am interpreting the changes and the reasons for the accurately.

**Review by spencerkclark — COMMENTED** [review]
I'm OK with the spirit of these changes, but I think there may be an issue with validating the `coarsening_factor` in the case when `fv_core_nml.write_coarse_restart_files` or `fv_core_nml.write_coarse_diagnostics` is set to `.true.`.  It may mean that we have to pass those namelist parameters in as arguments to `coarse_graining_init` and do the initial screen of the `coarsening_factor` (i.e. check that it is greater than zero) based on whether or not those are `.true.`.

**Comment by ofuhrer** [comment]
Closing this PR.

**Pull request closed without merging**